tests/ctl/core/test_utils.py

```diff
@@ -113,3 +113,12 @@ def test_new_file_is_dirty(self) -> None:
             file.write("test file")
         assert utils.git_is_dirty()
         os.remove(test_file)
+
+
+@pytest.mark.unit
+def test_repeatable_unique_key() -> None:
```

Comment:
are there any edge cases you can think of that might break this functionality?

The functionality is straight-forward and implemented in a transparent way, but still wondering if there might be anything else to consider for testing here

Reply:
I was thinking about it and the problem would really be if this were used with something other than datasets (or anything that doesn't have a host/name for us to use (which is out of scope for this change) and I couldn't really imagine how to test that. I liked this because it achieved the idea of the uuid being appended as well as being repeatable so two tests in one almost 👍🏽